src/resmom/linux/alps.c

```diff
@@ -3727,21 +3727,22 @@ inventory_loop_on_segments(basil_node_t *node, vnl_t *nv, char *arch,
 						goto bad_vnl;
 					if (accel->data.gpu->memory) {
 						attr = "resources_available.accelerator_memory";
-						if (totseg == 0) {
+						if (totseg == 0 || vnode_per_numa_node) {
 							snprintf(utilBuffer,
 								sizeof(utilBuffer),
 								"%umb",
 								accel->data.gpu->memory);
-						} else if (vnode_per_numa_node) {
-							snprintf(utilBuffer,
-								sizeof(utilBuffer),
-								"@%s_%ld_0",
-								mpphost, node->node_id);
+							if (vnode_per_numa_node && !totseg) {
+								snprintf(utilBuffer,
+									 sizeof(utilBuffer),
+									 "@%s_%ld_0",
+									 mpphost, node->node_id);
+							}
+							if (vn_addvnr(nv, vname, attr,
+								      utilBuffer, 0,
+								      0, NULL) == -1)
+								goto bad_vnl;
```

Comment:
> I don't understand why you changed this code from what had previously existed.

Ans. The previous implementation had a bug, which exist in 13.0.4xx as well.  Bug is  that, when vnode_per_numa_node is set to false, naccelerator and accelerator memory values are set wrong for the node which have more than one segment.

> Looks like now with vnode_per_numa_node set to true, the first NUMA node vnode will get accelerator memory set to @host_nodeid_0 (i.e. itself). 

Ans. Oops, i accidentally pushed the wrong code while rebasing. 
The condition should have been `if (vnode_per_numa_node && totseg != 0)`